plugins/services/src/js/components/forms/GeneralServiceFormSection.js

```diff
@@ -37,6 +46,89 @@ const containerRuntimes = {
 };
 
 class GeneralServiceFormSection extends Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {convertToPodModalOpen: false};
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  handleConvertToPod() {
+    this.props.onConvertToPod();
+    this.handleCloseConvertToPodModal();
+  }
+
+  handleCloseConvertToPodModal() {
+    this.setState({convertToPodModalOpen: false});
+  }
+
+  handleOpenConvertToPodModal() {
+    this.setState({convertToPodModalOpen: true});
+  }
+
+  getContainerSection() {
+    let {data = {}, errors} = this.props;
+
+    if (!(this.props.service instanceof PodSpec)) {
+      return (
+        <ContainerServiceFormSection
+          data={data}
+          errors={errors.container}
+          onAddItem={this.props.onAddItem}
+          onRemoveItem={this.props.onRemoveItem} />
+      );
+    }
+
+    const {containers = []} = data;
+    return containers.map((item, index) => {
+      return (
+        <FormGroupContainer key={index}
```

Comment:
The close button seems a little weirdly positioned in this view:
![](https://cl.ly/343x0R2R2I3y/Image%202016-12-20%20at%2013.58.26.png)